### PR TITLE
Update miekg/dns to v1.0.7

### PR DESCRIPTION
**Pull request by thaJeztah**
This dependency now uses SemVer, tagged releases. Also updates dependencies

- github.com/miekg/dns https://github.com/miekg/dns/compare/75e6e86cc601825c5dbcd4e0c209eab180997cd7...v1.0.7
- golang.org/x/crypto https://github.com/golang/crypto/compare/b080dc9a8c480b08e698fb1219160d598526310f...1a580b3eff7814fc9b40602fd35256c63b50f491
- golang.org/x/net https://github.com/golang/net/compare/5561cd9b4330353950f399814f427425c0a26fd2...0ed95abb35c445290478a5348a7b38bb154135fd

Libnetwork counterpart: https://github.com/docker/libnetwork/pull/2172. The x/crypto and x/net commits are the same as is used in https://github.com/moby/moby/pull/37059

### CODE REVIEW

**Comment by arm64b** [comment]
CI failure on `experimental`:

> 14:52:10 === RUN   TestCreateServiceConfigFileMode
14:52:22 --- FAIL: TestCreateServiceConfigFileMode (12.37s)
14:52:22 	daemon.go:285: [d94ae6ca8ae07] waiting for daemon to start
14:52:22 	daemon.go:317: [d94ae6ca8ae07] daemon started
14:52:22 	create_test.go:253: timeout hit after 10s: task count at 1 waiting for 0
14:52:22 	daemon.go:275: [d94ae6ca8ae07] exiting daemon

We may think to add `swarm.servicepoll` to override the default timeout value (10s)

**Review by vdemeester — APPROVED** [review]
LGTM :tiger:

**Review by vdemeester — APPROVED** [review]
LGTM :fire:

File: vendor.conf [hunk 1]
```diff
@@ -55,7 +55,7 @@ github.com/coreos/go-semver v0.2.0
 github.com/ugorji/go f1f1a805ed361a0e078bb537e4ea78cd37dcf065
 github.com/hashicorp/consul v0.5.2
 github.com/boltdb/bolt fff57c100f4dea1905678da7e90d92429dff2904
-github.com/miekg/dns 75e6e86cc601825c5dbcd4e0c209eab180997cd7
+github.com/miekg/dns v1.0.7
```
Comment (tiborvass) [inline vendor.conf]: @thaJeztah I know this is not consistent with other entries in vendor.conf but I would much rather have commits and use tags in a comment. If you want we can address this in a followup so we ensure it's consistent.

Comment (tiborvass) [reply vendor.conf]: Ok after discussing with @thaJeztah it looks like I'm the only one who's bothered by tags without shas. I'll see if I have enough energy to fight this in the broader Go ecosystem.

**Review by tiborvass — APPROVED** [review]
LGTM

**Pull request merged**